crates/router/src/connector/multisafepay/transformers.rs

```diff
@@ -0,0 +1,453 @@
+        });
+
+        let description = match _item.description.clone() {
+            Some(desc) => desc,
+            None => String::from("Default Description"),
+        };
+
+        let gateway_info = match _item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => {
+                GatewayInfo {
+                    card_number: Some(ccard.card_number.clone()),
+                    card_expiry_date: Some((format!("{}{}", ccard.card_exp_year.clone().expose(), ccard.card_exp_month.clone().expose())).parse::<i32>().unwrap()),
+                    card_cvc: Some(ccard.card_cvc.clone()),
+                    card_holder_name: None,
+                    flexible_3d: None,
+                    moto: None,
+                    term_url: None,
+                }
+            }
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "No Card Details".to_string(),
+            ))?,
+        };
+
+        Ok(Self {
+            _type: Some(String::from("direct")),
+            gateway: Gateway::Amex,
+            order_id: _item.payment_id.to_string(),
+            currency: _item.request.currency.to_string(),
+            amount: _item.request.amount,
+            description,
+            payment_options,
+            customer,
+            gateway_info: Some(gateway_info),
+            checkout_options: Some(CheckoutOptions { validate_cart: Some(false) }),
+            capture: None,
+            items: None,
+            recurring_model: None,
+            recurring_id: None,
+            days_active: Some(30),
+            seconds_active: Some(259200),
+            var1: None,
+            var2: None,
+            var3: None
+        })
+    }
+}
+
+//TODO: Fill the struct with respective fields
+// Auth Struct
+pub struct MultisafepayAuthType {
+    pub(super) api_key: String
+}
+
+impl TryFrom<&types::ConnectorAuthType> for MultisafepayAuthType  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::HeaderKey { api_key } = _auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType.into())
+        }
+    }
+}
+// PaymentsResponse
+//TODO: Append the remaining status flags
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum MultisafepayPaymentStatus {
+    Succeeded,
+    Failed,
+    #[default]
+    Pending,
+}
+
+impl From<MultisafepayPaymentStatus> for enums::AttemptStatus {
+    fn from(item: MultisafepayPaymentStatus) -> Self {
+        match item {
+            MultisafepayPaymentStatus::Succeeded => Self::Charged,
+            MultisafepayPaymentStatus::Failed => Self::Failure,
+            MultisafepayPaymentStatus::Pending => Self::AuthenticationPending,
```

Comment:
Is it always the case that if the payment is in processing state then it always requires a customer action?